File: Recursion-Backtracking/01-Print-Subsequences/print-subsequences.js
function printAllSubsequences(str){
  
  function findAllSubsequences(str, index, current){
    if(str.length === index){
      console.log(current);
      return;
    }
    findAllSubsequences(str, index + 1, current);
    findAllSubsequences(str, index + 1, current + str[index]);
  }
  
  findAllSubsequences(str, 0, "");
}


const str = "abc";
printAllSubsequences(str);

File: Recursion-Backtracking/01-Print-Subsequences/print-subsequences.cpp
#include <bits/stdc++.h>
using namespace std;

void findAllSubsequences(string str, int index, string& current){
  if(index == str.length()){
    cout << current << endl;
    return;
  }
  findAllSubsequences(str, index + 1, current);
  current.push_back(str[index]);
  findAllSubsequences(str, index + 1, current);
  current.pop_back();
}

int main() 
{
  string str = "abc";
  string current = "";
  findAllSubsequences(str, 0, current);
  return 0;
}